transport-native-unix-common-tests/src/main/java/io/netty/channel/unix/tests/DetectPeerCloseWithoutReadTest.java

```diff
@@ -0,0 +1,184 @@
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.SimpleChannelInboundHandler;
+import org.junit.Test;
+
+import java.net.InetSocketAddress;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+
+public abstract class DetectPeerCloseWithoutReadTest {
+    protected abstract EventLoopGroup serverGroup();
+    protected abstract EventLoopGroup clientGroup();
+    protected abstract Class<? extends ServerChannel> serverChannel();
+    protected abstract Class<? extends Channel> clientChannel();
+
+    @Test(timeout = 10000)
+    public void clientCloseWithoutServerReadIsDetected() throws InterruptedException {
+        EventLoopGroup serverGroup = null;
+        EventLoopGroup clientGroup = null;
+        Channel serverChannel = null;
+        try {
+            final CountDownLatch latch = new CountDownLatch(1);
+            final AtomicInteger bytesRead = new AtomicInteger();
+            final int expectedBytes = 100;
+            serverGroup = serverGroup();
+            clientGroup = clientGroup();
+            ServerBootstrap sb = new ServerBootstrap();
+            sb.group(serverGroup);
+            sb.channel(serverChannel());
+            sb.childOption(ChannelOption.AUTO_READ, false);
+            sb.childHandler(new ChannelInitializer<Channel>() {
+                @Override
+                protected void initChannel(Channel ch) throws Exception {
+                    ch.pipeline().addLast(new SimpleChannelInboundHandler<ByteBuf>() {
+                        @Override
+                        protected void channelRead0(ChannelHandlerContext ctx, ByteBuf msg) throws Exception {
+                            bytesRead.addAndGet(msg.readableBytes());
+                            // Because autoread is off, we call read to consume all data until we detect the close.
+                            ctx.read();
+                        }
+
+                        @Override
+                        public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+                            latch.countDown();
+                            ctx.fireChannelInactive();
+                        }
+                    });
+                }
+            });
+
+            serverChannel = sb.bind(new InetSocketAddress(0)).syncUninterruptibly().channel();
+
+            Bootstrap cb = new Bootstrap();
+            cb.group(serverGroup);
+            cb.channel(clientChannel());
+            cb.handler(new ChannelInitializer<Channel>() {
```

Comment:
nit: You can also just add a `ChannelInboundHandlerAdapter()`. This way you not need to implement any method.